ledger/src/blockstore.rs

```diff
@@ -156,6 +156,25 @@ impl PossibleDuplicateShred {
     }
 }
 
+enum WorkingEntry<T> {
+    Dirty(T), // Value has been modified with respect to the blockstore column
+    Clean(T), // Value matches what is currently in the blockstore column
+}
+
+impl<T> WorkingEntry<T> {
+    /// Returns a reference to the underlying value
+    fn value(&self) -> &T {
+        match self {
+            Self::Dirty(value) => value,
+            Self::Clean(value) => value,
+        }
+    }
```

Comment:
This could implement `AsRef` instead:
https://doc.rust-lang.org/std/convert/trait.AsRef.html